source/user/mcscf.rst

```diff
@@ -1,57 +1,299 @@
-.. 
-.. Woo, the total energy is even lower.  But the spin is contaminated.
+
+Picking an Active Space
+-----------------------
+In general, selecting an active space can be cumbersome and PySCF offers severals ways to facilitate this process.
+There are four strategies and all require that you specify the number of electrons and orbitals in the active space.
+
+.. warning::
+  We always advise users to visualize their chosen active orbitals before starting large/expensive calculations.
+  This involves dumping the MO coefficients to a ``molden`` file (see example :source:`examples/tools/02-molden.py`) and visualizing with your chosen program.
+  While there are many great softwares available to visualize orbitals, `JMol <http://jmol.sourceforge.net/>`_ is one of the easiest to use and is recommended for less experienced users.
+
+
+1) (Default) Specifying no additional information.
+  This is the most minimal strategy for selecting an active space and chooses orbitals (and electrons) around the fermi-level that match the number of orbitals and electrons specified by the user.
+  In most circumstances, this is not an ideal strategy and will lead to poor convergence or none at all.
+  For example:
+.. code-block:: python
+
+  ncas, nelecas = (6,8)
+  mycas = myhf.CASSCF(ncas, nelecas)
+
+
+2) Specifying the molecular orbital (MO) index of the active space orbitals you want. 
+  The user can "manually" selects the MO orbital indices (in a 1-based indexing scheme) and passes them to the ``sort_mo`` function.
+  See :source:`examples/mcscf/10-define_cas_space.py` for more details.
+
+.. code-block:: python
+
+  mycas = mcscf.CASSCF(myhf, 4, 4)
+  # Note sort_mo by default take the 1-based orbital indices.
+  mo = mycas.sort_mo([5,6,8,9])
+  mycas.kernel(mo)
+
+
+3) Specifying the number of orbitals in each symmetry group. 
+This strategy can occasionally be helpful when the initial guess orbitals are not easily identifiable.
+
+.. code-block:: python
+
+  mycas = mcscf.CASSCF(mf, 12, 12)
+  ncore = {'A1g':5, 'A1u':5}
+  ncas = {'A1g':2, 'A1u':2,'E1ux':1, 'E1uy':1, 'E1gx':1, 'E1gy':1,
+              'E2ux':1, 'E2uy':1, 'E2gx':1, 'E2gy':1}
+  mo = mcscf.sort_mo_by_irrep(mycas, mf.mo_coeff, ncas, ncore)
+  mycas.kernel(mo)
+
+A similar approach where we specify the electron occupations by irreducible representation is also possible by setting ``mycas.fcisolver.irrep_nelec``.
+
+.. code-block:: python
+
+  mycas = mcscf.CASSCF(myhf, 8, 8)
+  mycas.fcisolver.irrep_nelec = {"A1g": (2, 1), "A1u": (1, 1), "E1ux": (1, 1), "E1uy": (1, 0)}
+
+4) Use automated strategies (``avas`` and ``dmet_cas``) to pick an active space based on AO orbitals you're targeting.
+For more details, see :source:`examples/mcscf/43-avas.py` and :source:`examples/mcscf/43-dmet_cas.py`.
+
+.. code-block:: python
+
+  from pyscf.mcscf import avas
+  ao_labels = ['Fe 3d', 'Fe 4d', 'C 2pz']
+  ncas, nelecas, orbs = avas.avas(mf, ao_labels)
+  mycas = mcscf.CASSCF(mf, ncas, nelecas)
+
+  
```

Comment:
Is dmet-cas and AVAS just the same thing?

Reply:
I think AVAS actually runs localization.